src/node.rs

```diff
@@ -0,0 +1,219 @@
+use crate::helper::Identifier;
+use crate::kbucket::KbucketTable;
+use std::collections::HashMap;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use tokio::io;
+use tokio::net::UdpSocket;
+
+#[derive(Debug)]
+pub enum FindNodeResult {
+    Found(Option<TableRecord>),
+    NotFound(Vec<Option<TableRecord>>),
+}
+
+#[derive(Debug)]
+pub enum Search {
+    Success(usize, usize),
+    Failure(usize, usize),
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct TableRecord {
+    pub node_id: Identifier,
+    pub ip_address: Ipv4Addr,
+    pub udp_port: u16,
+    pub socket_addr: SocketAddrV4,
```

Comment:
I would only store ip and port, not the `socket_addr` as well

Reply:
For whatever reason the stack overflow error solved itself after removing socket_addr from the Table Record...

Reply:
I'd either store a `SocketAddrV4` or the pair, but you don't really need both